From 4d40b06b858dd394905941b87fdf95925813d786 Mon Sep 17 00:00:00 2001
From: Stewart Smith <stewart@linux.vnet.ibm.com>
Date: Tue, 8 Dec 2015 12:05:04 +1100
Subject: [PATCH 1/3] linker/genlist: use stderr for displaying error messages

Since genlist is used in makefiles and stdout redirected,
if there is an error, you will not see it in make output at
all, instead having things fail in weird ways.

Use fprintf for stderr rather than to stdout to avoid

Signed-off-by: Stewart Smith <stewart@linux.vnet.ibm.com>
---
 src/build/linker/genlist.C | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/build/linker/genlist.C b/src/build/linker/genlist.C
index 69b52a1..11bb583 100644
--- a/src/build/linker/genlist.C
+++ b/src/build/linker/genlist.C
@@ -109,7 +109,7 @@ int main(int argc, char** argv)
     g_crossPrefix = getenv("CROSS_PREFIX");
     if (NULL == g_crossPrefix)
     {
-        printf("Environment variable CROSS_PREFIX not set.\n");
+	fprintf(stderr, "Environment variable CROSS_PREFIX not set.\n");
         exit(-1);
     }
     g_crossPrefix = strdup(g_crossPrefix);
@@ -155,7 +155,7 @@ int main(int argc, char** argv)
 
 void print_usage()
 {
-    printf("genlist <image>\n");
+    fprintf(stderr, "genlist <image>\n");
     exit(-1);
 }
 
@@ -176,7 +176,7 @@ void parse_modinfo_file(const string& i_image)
     FILE* modinfo_file = fopen(modinfo_name.c_str(), "r");
     if (NULL == modinfo_file)
     {
-        printf("Unable to open modinfo file.\n");
+        fprintf(stderr, "Unable to open modinfo file.\n");
         exit(-1);
     }
 
@@ -217,7 +217,7 @@ void parse_syms_file(const string& i_image)
     FILE* syms_file = fopen(syms_name.c_str(), "r");
     if (NULL == syms_file)
     {
-        printf("Unable to open syms file.\n");
+        fprintf(stderr, "Unable to open syms file.\n");
         exit(-1);
     }
 
-- 
2.5.0

